### PR TITLE
fix: positions should be zero when market is settled

### LINKED ISSUE #5998: [graphql] Positions should not have non-zero unrealisedPNL or openVolume
# Problem encountered
A user has a position in a market that has settled. GraphQL API is returning non-zero `openVolume` and `unrealisedPNL`

Query run on testnet node 10:
```graphql
{
 party(id:"3aec463efd76df0c584b5f90180acd2a3d5f3d3e1c3d7b0d235337d77be6cce9"){
    positionsConnection{
      edges {
        node {
          market { name state }
      	  realisedPNL
          unrealisedPNL
          openVolume
        }
      }
    }
  }
}
```
```json
{
  "data": {
    "party": {
      "positionsConnection": {
        "edges": [
          {
            "node": {
              "market": {
                "name": "OP/USD AUG-SEP22 - Incentive.",
                "state": "STATE_ACTIVE"
              },
              "realisedPNL": "0",
              "unrealisedPNL": "-380",
              "openVolume": "2"
            }
          },
          {
            "node": {
              "market": {
                "name": "OP/USD AUG-SEP22 - Incentive",
                "state": "STATE_SETTLED"
              },
              "realisedPNL": "0",
              "unrealisedPNL": "0",
              "openVolume": "0"
            }
          },
          {
            "node": {
              "market": {
                "name": "OP/USD AUG-SEP22",
                "state": "STATE_SETTLED"
              },
              "realisedPNL": "24",
              "unrealisedPNL": "52",
              "openVolume": "4"
            }
          },
          {
            "node": {
              "market": {
                "name": "BTCUSD (1 days) - 8 prices",
                "state": "STATE_SETTLED"
              },
              "realisedPNL": "-8111953085",
              "unrealisedPNL": "0",
              "openVolume": "0"
            }
          }
        ]
      }
    }
  }
}
```

**Note** Diving deeper in to the OP/USD AUG-SEP22 market, all of the user's orders are finalised (filled/stopped/cancelled/rejected) so it's likely to be the positions API having not done all the tidyup, as per @barnabee's note below.

# Expected behaviour
All orders are closed out and settled. `openVolume` is zero. `unrealisedPNL` is 0 and `realisedPNL` is set appropriately

# Steps to reproduce

## Manual
Steps to reproduce the behaviour manually:
1. Propose a market
2. Vote the market in
3. Provide liquidity so the market gets out of opening auction
4. Trade a bit
5. Settle the market (JSON oracle, unsure if this is important)
6. Perform the query above

> So core probably emits a “market settled” event and then deletes the entire data structure. No need to update a position to zero if you're going to delete it.

> Data node on the other hand cares about historic information so it needs to decide the position is now zero, either by interpreting the “market settled” event in a way that updates all positions on the market to zero, or by asking core to emit more of some other event that would cause the position to be updated to zero.

### PR DESCRIPTION
**Pull request by guoguojin**
Closes #5998

### CODE REVIEW

File: datanode/entities/position.go [hunk 1]
```diff
@@ -111,6 +117,17 @@ func (p *Position) UpdateWithSettleDestressed(e settleDestressed) {
 	p.TxHash = TxHash(e.TxHash())
 }
 
+func (p *Position) UpdateWithSettleMarket(e settleMarket) {
+	markPriceDec := num.DecimalFromUint(e.SettledPrice())
+	openVolumeDec := num.DecimalFromInt64(p.OpenVolume)
+
+	unrealisedPnl := openVolumeDec.Mul(markPriceDec.Sub(p.AverageEntryPrice)).Div(e.PositionFactor())
+	p.RealisedPnl = p.RealisedPnl.Add(unrealisedPnl)
+	p.UnrealisedPnl = decimal.Zero
```
Comment (EVODelavega) [inline datanode/entities/position.go:126]: Not a fan of this, TBH... Of course we have `type Decimal = decimal.Decimal` in the `num` package, but if we ever were to swap out the underlying type, having the underlying type bleed in to the code base would be an issue

Comment (jeremyletang) [reply datanode/entities/position.go:126]: that's on the datanode though, it initially started with using the decimal package directly.

File: core/execution/market.go [hunk 2]
```diff
@@ -3232,6 +3232,7 @@ func (m *Market) settlementPriceWithLock(ctx context.Context) {
 
 			// send the market data with all updated stuff
 			m.broker.Send(events.NewMarketDataEvent(ctx, m.GetMarketData()))
+			m.broker.Send(events.NewMarketSettled(ctx, m.GetID(), m.timeService.GetTimeNow().UnixNano(), m.markPrice, m.positionFactor))
 		} else {
```
Comment (EVODelavega) [inline core/execution/market.go:3236]: nit-pick: this `else` seems pointless. Return early, and remove the `else`

File: datanode/entities/position.go (hunk 1 above)
Comment (guoguojin) [reply datanode/entities/position.go]: Thanks, I've updated the other places it was used in the other positions methods too with this commit https://github.com/vegaprotocol/vega/pull/6262/commits/dd3080400fa705880944db06480782ac0a45ee6b. It'll be used a lot in other places in data-node so maybe we should raise a low priority ticket to clean that up at some point.

File: core/execution/market.go (hunk 2 above)
Comment (guoguojin) [reply core/execution/market.go]: Addressed with this commit https://github.com/vegaprotocol/vega/pull/6262/commits/dd3080400fa705880944db06480782ac0a45ee6b

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**